src/compiler/utilities.ts

```diff
@@ -2726,8 +2726,9 @@ namespace ts {
         return forEach(supportedTypeScriptExtensions, extension => fileExtensionIs(fileName, extension));
     }
 
-    export function hasTypeScriptFileExtensionNonDts(fileName: string) {
-        return forEach(supportedTypeScriptExtensionsNonDts, extension => fileExtensionIs(fileName, extension));
+    /** Return ".ts" or ".tsx" if that is the extension. */
+    export function tryExtractTypeScriptExtensionNonDts(fileName: string): string | undefined {
+        return find(supportedTypeScriptExtensionsNonDts, extension => fileExtensionIs(fileName, extension));

```

Comment:
If people add a `.d.ts` extension they may be doing unnecessary work, but it won't cause runtime errors because declaration references don't translate to imports in the output. In contrast, a `.ts` extension will either fail to import (because the output file has a `.js` extension) or ask the JS engine to load typescript code.